webapp/src/pages/ResponsePage/ResponsePage.js

```diff
@@ -0,0 +1,211 @@
+        }
+
+        return html
+    }
+
+
+
+    // Render Questions 
+    renderQuestions(section) {
+        let questions = section.questions.map(q => {
+            return <div key={q.id} className="question-answer-block">
+                <p>{q.headline}</p>
+                <h6>{this.renderAnswer(q.id, q.type)}</h6>
+            </div>
+        })
+        return questions
+    }
+
+
+
+    // Render Answers 
+    renderAnswer(id, type) {
+        const applicationData = this.state.applicationData;
+        const baseUrl = process.env.REACT_APP_API_URL;
+        let answers;
+
+        applicationData.answers.forEach(a => {
+            if (a.question_id == id) {
+                formatAnswer(a, type)
+            }
+        })
+
+        // format aswerers 
+        function formatAnswer(a, type) {
+            // file
+            if (type == "file") {
+                answers = <a className="answer file" key={a.value} target="_blank" href={baseUrl + "/api/v1/file?filename=" + a.value}>{a.value}</a>
+            }
+            // multi-file
+            if (type == "multi-file") {
+                let files = [];
+                a.value.forEach((file => {
+                    file ? files.push(
+                        <div key={a.headline}><a key={a.headline} target="_blank" href={baseUrl + "/api/v1/file?filename=" + file} className="answer">{a.value}</a></div>
+                    )
+                        :
+                        console.log(`${a.question_id} contains no value`)
+                }))
+                answers = <div key={a.headline}>{files}</div>
+            }
+            // choice
+            if (type.includes("choice")) {
+                let choices = [];
+                a.options.forEach(opt => {
+                    a.value.forEach(val => {
+                        if (val.label == opt.label) {
+                            choices.push(<div key={opt.label}><label className="answer">{opt.label}</label></div>)
+                        }
+                    })
+                })
+                answers = <div key={choices}>{choices}</div>
+            }
+            // text
+            if (type.includes("text")) {
+                answers = <div key={a.headline}><p className="answer">{a.value}</p></div>
+            }
+        }
+
+        return answers
+    }
+
+
+    render() {
+        const { applicationForm, applicationData } = this.state
+        const applicationStatus = this.applicationStatus();
+        const renderSections = this.renderSections();
+
+        // Translation
+        const t = this.props.t;
+
+        return (
+            <div className="table-wrapper">
+                {applicationData &&
+                    <div className="headings-lower">
+                        <div>
+                            <div className="user-details"> <p>{applicationData.user_title}</p> </div>
+                            <div className="user-details"> <p>{applicationData.firstname}</p> </div>
```

Comment:
This seems overly complicated? Why not just `<h3>{applicationData.user_title} {applicationData.firstname} {applicationData.lastname}</h3>` ?

Reply:
Resolved